### PR TITLE
PivotGrid: Update Field Chooser demo descriptions

### PR DESCRIPTION
**Pull request by RomanTsukanov**

### CODE REVIEW

**Review by novsstation — DISMISSED** [review]

File: JSDemos/Demos/PivotGrid/IntegratedFieldChooser/description.md [hunk 1]
```diff
@@ -1 +1,29 @@
...
+
+You can group All Fields for hierarchical display. Specify the same [displayFolder](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/#displayFolder) for the fields that should be grouped together. In this demo, the hierarchy is built on the server. 
+
+Users can drag and drop fields between the sections. When users move a field to Row, Column, Data, or Filter Fields, it is added to the corresponding pivot grid area. To do the same programmatically, specify the field's [area](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/#area) property. If a field is intended to be moved to the Data Fields section only, enable the field's [isMeasure](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/#isMeasure) property, and vice versa: disable the **isMeasure** property for fields that should never be in the Data Fields section. In this demo, these restrictions are specified on the server.
```
Comment (AlexJameson) [inline JSDemos/Demos/PivotGrid/IntegratedFieldChooser/description.md]: > When users move a field to Row, Column, Data, or Filter Fields, it is added to the corresponding pivot grid area.

Would not it be more clear to say 'to the Row, Column, Data, or Filter Field sections, the PivotGrid adds this field to the corresponding area.'

```suggestion
Users can drag and drop fields between the sections. When users move a field to the Row, Column, Data, or Filter Fields section, the PivotGrid adds this field to the corresponding area. To do the same programmatically, specify the field's [area](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/#area) property. If a field is intended to be moved to the Data Fields section only, enable the field's [isMeasure](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/#isMeasure) property, and vice versa: disable the **isMeasure** property for fields that should never be in the Data Fields section. In this demo, these restrictions are specified on the server.
```

File: JSDemos/Demos/PivotGrid/IntegratedFieldChooser/description.md [hunk 2]
```diff
@@ -1 +1,29 @@
...
+
+After a user finishes moving fields between sections, the changes are applied immediately or after the user clicks OK. Use the **fieldChooser**.[applyChangesMode](/Documentation/ApiReference/UI_Components/dxPivotGrid/Configuration/fieldChooser/#applyChangesMode) property to set the desired mode. In this demo, you can change this property at runtime.
+
+### Enable Search
+
+Users can search the All Fields section if you enable the **fieldChooser**.[allowSearch](/Documentation/ApiReference/UI_Components/dxPivotGrid/Configuration/fieldChooser/#allowSearch) property as shown in this demo. In addtion, you can specify the [searchTimeout](/Documentation/ApiReference/UI_Components/dxPivotGrid/Configuration/fieldChooser/#searchTimeout) property to delay the search.
```
Comment (AlexJameson) [inline JSDemos/Demos/PivotGrid/IntegratedFieldChooser/description.md]: >Users can search the All Fields section

Should it be search **in** the All Fields section?

```suggestion
Users can search in the All Fields section if you enable the **fieldChooser**.[allowSearch](/Documentation/ApiReference/UI_Components/dxPivotGrid/Configuration/fieldChooser/#allowSearch) property as shown in this demo. In addtion, you can specify the [searchTimeout](/Documentation/ApiReference/UI_Components/dxPivotGrid/Configuration/fieldChooser/#searchTimeout) property to delay the search.
```

Comment (AlexJameson) [inline JSDemos/Demos/PivotGrid/IntegratedFieldChooser/description.md]:
```suggestion
Users can search the All Fields section if you enable the **fieldChooser**.[allowSearch](/Documentation/ApiReference/UI_Components/dxPivotGrid/Configuration/fieldChooser/#allowSearch) property as shown in this demo. In addition, you can specify the [searchTimeout](/Documentation/ApiReference/UI_Components/dxPivotGrid/Configuration/fieldChooser/#searchTimeout) property to delay the search.
```

File: JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md [hunk 3]
```diff
@@ -1 +1,14 @@
...
+
+1. *(Optional)* **Apply field changes on demand**           
+When users move fields within the field chooser, the changes are applied to the connected PivotGrid instantly. To change this behavior, you can set the [applyChangesMode](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Configuration/#applyChangesMode) to *"onDemand"*. In this mode, the changes are applied/canceled only when you call the [applyChanges()](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Methods/#applyChanges)/[cancelChanges()](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Methods/#cancelChanges) method or set a new [state](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Configuration/#state). You can add a UI for these methods. For example, in this demo, they are bound to the Apply and Cancel buttons. Select "onDemand" from the Apply Changes Mode drop-down menu to display these buttons.
```
Comment (AlexJameson) [inline JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md]: >You can add a UI for these methods.

Not sure that this sentence is valid. I would rephrase it as follows: 'You can call these methods from the UI' or something similar. 

However, if you are sure that this sentence is OK, just skip this comment.

**Review by AlexJameson — DISMISSED** [review]

File: JSDemos/Demos/PivotGrid/IntegratedFieldChooser/description.md [hunk 4]
```diff
@@ -1 +1,29 @@
-The field chooser is&nbsp;a&nbsp;pivot grid element that allows you to&nbsp;manage the displayed fields. You can invoke it&nbsp;by&nbsp;clicking the **Field Chooser** icon at&nbsp;the top-left corner or&nbsp;by&nbsp;selecting the *"Show Field Chooser"* command on&nbsp;the PivotGrid's context menu. Changes made in&nbsp;the field chooser can be&nbsp;applied to&nbsp;the PivotGrid in&nbsp;instant or&nbsp;on-demand mode. In&nbsp;the latter case, the field chooser is&nbsp;supplied with the&nbsp;**OK** and **Cancel** buttons.
\ No newline at end of file
+The field chooser allows users to manage displayed pivot grid fields. You can configure the field chooser integrated into the PivotGrid or [use it as a standalone component](https://js.devexpress.com/Demos/WidgetsGallery/Demo/PivotGrid/StandaloneFieldChooser/). This example demonstrates the integrated field chooser. To open the field chooser window, click the icon in the top-left corner or right-click a row or column header and select Show Field Chooser from the appeared context menu.
```
Comment (Abadzhev) [inline JSDemos/Demos/PivotGrid/IntegratedFieldChooser/description.md]: I would remove "displayed" and "appeared". Does "displayed" mean "visible"? I guess visibility changes are part of the management... And context menu is fine without "appeared".

```suggestion
The field chooser allows users to manage pivot grid fields. You can configure the field chooser integrated into the PivotGrid or [use it as a standalone component](https://js.devexpress.com/Demos/WidgetsGallery/Demo/PivotGrid/StandaloneFieldChooser/). This example demonstrates the integrated field chooser. To open the field chooser window, click the icon in the top-left corner or right-click a row or column header and select Show Field Chooser from the context menu.
```

File: JSDemos/Demos/PivotGrid/IntegratedFieldChooser/description.md [hunk 5]
```diff
@@ -1 +1,29 @@
...
+
+- Row Fields
+- Column Fields
+- Data Fields
+- Filter Fields
+- All Fields
+
+You can use the **fieldChooser**.[layout](/Documentation/ApiReference/UI_Components/dxPivotGrid/Configuration/fieldChooser/#layout) property to arrange the sections in different ways.
+
+All Fields include fields declared in the [fields[]](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/) array and auto-generated fields as shown in this demo. If you want to hide the auto-generated fields, disable the [dataSource](/Documentation/ApiReference/UI_Components/dxPivotGrid/Configuration/#dataSource).[retrieveFields](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/#retrieveFields) property. You can also hide any particular field if you disable its [visible](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/#visible) property.
```
Comment (Abadzhev) [inline JSDemos/Demos/PivotGrid/IntegratedFieldChooser/description.md]:
```suggestion
The "All Fields" section includes fields declared in the [fields[]](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/) array and auto-generated fields as shown in this demo. If you want to hide the auto-generated fields, disable the [dataSource](/Documentation/ApiReference/UI_Components/dxPivotGrid/Configuration/#dataSource).[retrieveFields](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/#retrieveFields) property. You can also hide any particular field if you disable its [visible](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/#visible) property.
```

File: JSDemos/Demos/PivotGrid/IntegratedFieldChooser/description.md [hunk 6]
```diff
@@ -1 +1,29 @@
...
+
+You can group All Fields for hierarchical display. Specify the same [displayFolder](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/#displayFolder) for the fields that should be grouped together. In this demo, the hierarchy is built on the server.
```
Comment (Abadzhev) [inline JSDemos/Demos/PivotGrid/IntegratedFieldChooser/description.md]:
```suggestion
You can enable hierarchical display in the All Fields section. Specify the same [displayFolder](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/#displayFolder) for the fields that should be grouped together. In this demo, the hierarchy is built on the server. 
```

File: JSDemos/Demos/PivotGrid/IntegratedFieldChooser/description.md [hunk 7]
```diff
@@ -1 +1,29 @@
...
+
+Users can drag and drop fields between the sections. When users move a field to the Row, Column, Data, or Filter Fields section, the PivotGrid adds this field to the corresponding area. To do the same programmatically, specify the field's [area](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/#area) property. If a field is intended to be moved to the Data Fields section only, enable the field's [isMeasure](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/#isMeasure) property, and vice versa: disable the **isMeasure** property for fields that should never be in the Data Fields section. In this demo, these restrictions are specified on the server.
+
+After a user finishes moving fields between sections, the changes are applied immediately or after the user clicks OK. Use the **fieldChooser**.[applyChangesMode](/Documentation/ApiReference/UI_Components/dxPivotGrid/Configuration/fieldChooser/#applyChangesMode) property to set the desired mode. In this demo, you can change this property at runtime.
```
Comment (Abadzhev) [inline JSDemos/Demos/PivotGrid/IntegratedFieldChooser/description.md]: Once I read "are applied immediately", I assume that's the only option. But then you say "or after"... This sounds a bit confusing. If I use "either ... or ... ", readers will anticipate two different options.

```suggestion
After a user finishes moving fields between sections, the changes can either be applied immediately or after the user clicks OK. Use the **fieldChooser**.[applyChangesMode](/Documentation/ApiReference/UI_Components/dxPivotGrid/Configuration/fieldChooser/#applyChangesMode) property to set the desired mode. In this demo, you can change this property at runtime.
```

File: JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md [hunk 8]
```diff
@@ -1 +1,14 @@
-The standalone field chooser is&nbsp;the&nbsp;PivotGridFieldChooser component that shares the data source with the PivotGrid and has three different layouts. Active layout changes can be&nbsp;applied to&nbsp;the PivotGrid in&nbsp;instant or&nbsp;on-demand mode.
\ No newline at end of file
+Along with a field chooser [integrated into the PivotGrid](https://js.devexpress.com/Demos/WidgetsGallery/Demo/PivotGrid/IntegratedFieldChooser/), DevExtreme includes a standalone [PivotGridFieldChooser](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/) component. Unlike its integrated counterpart, the standalone field chooser can be displayed on a page permanently.
```
Comment (Abadzhev) [inline JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md]: "Displayed permanently" sounds a bit weird to me. Partly because of passive voice, partly because the word "permanently" doesn't seem right. 

```suggestion
Along with a field chooser [integrated into the PivotGrid](https://js.devexpress.com/Demos/WidgetsGallery/Demo/PivotGrid/IntegratedFieldChooser/), DevExtreme includes a standalone [PivotGridFieldChooser](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/) component. Unlike its integrated counterpart, the standalone field chooser can remain visible on the page at all times.
```

File: JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md (hunk 3 above)
Comment (Abadzhev) [reply JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md]:
```suggestion
When users move fields within the field chooser, the changes are applied to the connected PivotGrid instantly. To change this behavior, you can set the [applyChangesMode](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Configuration/#applyChangesMode) to *"onDemand"*. In this mode, the changes are applied/canceled only when you call the [applyChanges()](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Methods/#applyChanges)/[cancelChanges()](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Methods/#cancelChanges) method or set a new [state](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Configuration/#state). This demo adds custom Apply and Cancel buttons that call the methods mentioned above. Select "onDemand" from the Apply Changes Mode drop-down menu to display these buttons.
```

File: JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md [hunk 9]
```diff
@@ -1 +1,14 @@
...
+
+The standalone and integrated field choosers share features. For example, both field choosers include three [layouts](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Configuration/#layout). You can spot the difference between them in this demo. For information about other available features, refer to the [Integrated Field Chooser](https://js.devexpress.com/Demos/WidgetsGallery/Demo/PivotGrid/IntegratedFieldChooser/) demo.
```
Comment (Abadzhev) [inline JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md]: No explicit information on how to "spot differences". 

```suggestion
The standalone and integrated field choosers share features. For example, both field choosers include three [layouts](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Configuration/#layout). You can use the Radio Buttons to switch between them and review differences. For information about other available features, refer to the [Integrated Field Chooser](https://js.devexpress.com/Demos/WidgetsGallery/Demo/PivotGrid/IntegratedFieldChooser/) demo.
```

File: JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md [hunk 10]
```diff
@@ -1 +1,14 @@
...
+
+1. *(Optional)* **Apply field changes on demand**           
+When users move fields within the field chooser, the changes are applied to the connected PivotGrid instantly. To change this behavior, you can set the [applyChangesMode](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Configuration/#applyChangesMode) to *"onDemand"*. In this mode, the changes are applied/canceled only when you call the [applyChanges()](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Methods/#applyChanges)/[cancelChanges()](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Methods/#cancelChanges) method or set a new [state](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Configuration/#state). You can add a UI for these methods. For example, in this demo, they are bound to the Apply and Cancel buttons. Select "onDemand" from the Apply Changes Mode drop-down menu to display these buttons.
```
Comment (dirk-pieterse) [inline JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md]:
```suggestion
When users move fields within the field chooser, the changes are applied to the connected PivotGrid instantly. To change this behavior, set the [applyChangesMode](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Configuration/#applyChangesMode) to *"onDemand"*. In this mode, the changes are applied/canceled only when you call the [applyChanges()](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Methods/#applyChanges)/[cancelChanges()](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Methods/#cancelChanges) method or set a new [state](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Configuration/#state). You can add a UI for these methods. For example, in this demo, they are bound to the Apply and Cancel buttons. Select "onDemand" from the Apply Changes Mode drop-down menu to display these buttons.
```

File: JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md [hunk 11]
```diff
@@ -1 +1,14 @@
...
+
+The standalone and integrated field choosers share features. For example, both field choosers include three [layouts](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Configuration/#layout). You can use the radio buttons to switch between them and review differences. For information about other available features, refer to the [Integrated Field Chooser](https://js.devexpress.com/Demos/WidgetsGallery/Demo/PivotGrid/IntegratedFieldChooser/) demo.
```
Comment (dirk-pieterse) [inline JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md]:
```suggestion
The standalone and integrated field choosers share features. For example, both field choosers include three [layouts](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/Configuration/#layout). You can use the radio buttons to switch between field choosers and review the differences. For information about other available features, refer to the [Integrated Field Chooser](https://js.devexpress.com/Demos/WidgetsGallery/Demo/PivotGrid/IntegratedFieldChooser/) demo.
```

File: JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md [hunk 12]
```diff
@@ -1 +1,14 @@
-The standalone field chooser is&nbsp;the&nbsp;PivotGridFieldChooser component that shares the data source with the PivotGrid and has three different layouts. Active layout changes can be&nbsp;applied to&nbsp;the PivotGrid in&nbsp;instant or&nbsp;on-demand mode.
\ No newline at end of file
+Along with a field chooser [integrated into the PivotGrid](https://js.devexpress.com/Demos/WidgetsGallery/Demo/PivotGrid/IntegratedFieldChooser/), DevExtreme includes a standalone [PivotGridFieldChooser](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/) component. Unlike its integrated counterpart, the standalone field chooser can remain visible on the page at all times.
```
Comment (dirk-pieterse) [inline JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md]:
```suggestion
DevExtreme also includes a standalone [PivotGridFieldChooser](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/) component with the [integrated](https://js.devexpress.com/Demos/WidgetsGallery/Demo/PivotGrid/IntegratedFieldChooser/) field chooser. Unlike its integrated counterpart, the standalone field chooser can remain visible on the page at all times.
```

**Review by dirk-pieterse — DISMISSED** [review]

File: JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md (hunk 12 above)
Comment (RomanTsukanov) [reply JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md]: > a standalone [PivotGridFieldChooser](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/) component with the [integrated](https://js.devexpress.com/Demos/WidgetsGallery/Demo/PivotGrid/IntegratedFieldChooser/) field chooser

The standalone field chooser doesn't include the integrated field chooser. It's DevExtreme that includes a standalone field chooser and a field chooser integrated into the PivotGrid.

Comment (dirk-pieterse) [reply JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md]: Why not just say 'DevExtreme also includes a standalone [PivotGridFieldChooser](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/) component'?

Comment (RomanTsukanov) [reply JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md]: We have two different demos for the integrated and standalone field choosers. Those who enter our website from the standalone field chooser demo will be confused by this "also". I quess the following version would work:

```suggestion
DevExtreme includes a standalone [PivotGridFieldChooser](/Documentation/ApiReference/UI_Components/dxPivotGridFieldChooser/) component. Unlike its [integrated counterpart](https://js.devexpress.com/Demos/WidgetsGallery/Demo/PivotGrid/IntegratedFieldChooser/), the standalone field chooser can remain visible on the page at all times.
```

Comment (dirk-pieterse) [reply JSDemos/Demos/PivotGrid/StandaloneFieldChooser/description.md]: Ok
